src/gui/qgisgui.h

```diff
@@ -83,14 +83,12 @@ namespace QgisGui
   QPair<QString, QString> GUI_EXPORT getSaveAsImageName( QWidget * theParent, QString theMessage, QString defaultFilename = QString::null );
 
   /**
-    Convenience function for readily creating file filters.
-
-    Given a long name for a file filter and a regular expression, return
-    a file filter string suitable for use in a QFileDialog::OpenFiles()
-    call.  The regular express, glob, will have both all lower and upper
-    case versions added.
-  */
-  QString createFileFilter_( QString const &longName, QString const &glob );
+   * Create file filters suitable for use with QFileDialog
+   *
+   * @param format extension e.g. "png"
+   * @return QString e.g. "PNG format (*.png, *.PNG)"
+   */
+  QString createFileFilter_( QString const &format );
```

Comment:
I've merged upstream, and added back the original definition. I believe I've addressed your concerns around breaking the API.

This is all new to me but GUI_EXPORT is especially foreign. I'm assuming I need it for the new `QString createFileFilter_( QString const &format )` so I've added that. Let me know if I'm off base, I'm new here :wink: